ext/standard/proc_open.c

```diff
@@ -1414,10 +1460,10 @@ PHP_FUNCTION(proc_open)
 	}
 
 	if (1) {
-		RETVAL_RES(zend_register_resource(proc, le_proc_open));
 	} else {
 exit_fail:
 		_php_free_envp(env);
+		zval_ptr_dtor(return_value);
 		RETVAL_FALSE;
 	}
```

Comment:
Am I blind, or can the jump to ``exit_fail`` only happen *prior* to allocating the object?

Reply:
uhh, nice catch! Probably I put `zval_ptr_dtor(return_value);` here when the allocation used to be on an earlier line. 🤔